### PR TITLE
Reorg the monitoring task section

### LINKED ISSUE #32644: Reorg the "monitoring, logging and debugging" subsection.
The current organization of the https://kubernetes.io/docs/tasks/debug-application-cluster/ section is a mixture of several topics targeting different group of audience. There are topics related to debugging applications, there are also topics on debugging node, cluster level issues. There are three subsections about 'troubleshooting'. This is confusing and making it difficult to find the right information user wants to read.

**Proposed solution**

- Add a "debugging applications" subsection and migrate all relate pages there, this subsection targets application developers;
- Add a "debugging cluster" subsection and migrate all related pages there, this subsection targets cluster admins.
- Split other pages properly, and organize the contents accordingly.

### PR DESCRIPTION
**Pull request by psschwei**
Signed-off-by: Paul S. Schweigert <paulschw@us.ibm.com>

Fixes #32644 

Reorganizes the "Monitoring, Logging, and Debugging" subsection into four main areas:

* A general overview, with links for additional help
* A section on debugging applications
* A section on debugging clusters
* A final section on more general monitoring/logging/observability topics

I also removed some pages that duplicated others / moved some sections into different pages to help consolidate the information.

### CODE REVIEW

File: content/en/docs/tasks/debug-application-cluster/applications/_index.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/applications/_index.html:1]: this should be _index.md ?

Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/applications/_index.html:1]: move this whole section one level up, and name it as "debug-applications"?

File: content/en/docs/tasks/debug-application-cluster/applications/debug-application.md [hunk 2]
```diff
@@ -2,8 +2,9 @@
 reviewers:
 - mikedanese
 - thockin
-title: Troubleshoot Applications
+title: Debug Pods
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/applications/debug-application.md:5]: If this page is about debug pods, then better name the file as "debug-pods.md".

File: content/en/docs/tasks/debug-application-cluster/monitoring/monitor-node-health.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ content_type: task
 reviewers:
 - Random-Liu
 - dchen1107
+weight: 20
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/monitoring/monitor-node-health.md:7]: Maybe put this file and following files under `content/docs/tasks/debug-cluster/`?

File: content/en/docs/tasks/debug-application-cluster/applications/_index.html (hunk 1 above)
Comment (psschwei) [reply content/en/docs/tasks/debug-application-cluster/applications/_index.html:1]: If we move the application / cluster sections up a level, what about the `troubleshooting.md` doc? It's sort of a general info doc, so probably doesn't need its own high-level section, so maybe it makes sense to organize like this:

```
tasks/
   |--- troubleshooting.md
   |--- debug-application/
      |--- debug-pods.md
      |--- (other application docs...)
   |--- debug-cluster/
      |--- debug-cluster.md
      |--- (all the docs currently under monitoring/...)
```

Open to other suggestions as well, that's just what initially came to mind...

Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/applications/_index.html]: We may need to investigate the contents in `troubleshooting.md` and split them accordingly. Troubleshooting a cluster is completely a different topic than troubleshooting an application or a Pod.

Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/applications/_index.html]: Checked the contents in the `troubleshooting.md` file. I agree with you that it is a generic page. So your proposal makes good sense.

File: content/en/docs/tasks/debug/_index.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Debugging"
```
Comment (tengqm) [inline content/en/docs/tasks/debug/_index.md:2]: Maybe we should keep the current title "Monitoring, Logging, and Debugging" so that users can still quickly find the topic they want.

File: content/en/docs/tasks/debug/debug-application/debug-stateful-set.md [hunk 5]
```diff
@@ -9,6 +9,7 @@ reviewers:
 - smarterclayton
 title: Debug a StatefulSet
 content_type: task
+weight: 30
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-application/debug-stateful-set.md:12]: I'd suggest we name this page as "debug-statefulset.md".

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 6]
```diff
@@ -0,0 +1,308 @@
+---
+reviewers:
+- davidopp
+title: Troubleshoot Clusters
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:4]:
```suggestion
title: Troubleshooting Clusters
```
For consistency?

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+the [application troubleshooting guide](/docs/tasks/debug-application-cluster/applications/debug-application) for tips on application debugging.
+You may also visit [troubleshooting document](/docs/tasks/debug-application-cluster/troubleshooting/) for more information.
+
+<!-- body -->
+
+## Listing your cluster
+
+The first thing to debug in your cluster is if your nodes are all registered correctly.
+
+Run
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:23]:
```suggestion
The first thing to debug in your cluster is if your nodes are all registered correctly.
Run the following command:
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+    systemUUID: aa829ca9-73d7-064d-9019-df07404ad448
+```
+
+
+## Looking at logs
+
+For now, digging deeper into the cluster requires logging into the relevant machines.  Here are the locations
+of the relevant log files.  (note that on systemd-based systems, you may need to use `journalctl` instead)
+
+### Master
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:223]:
```suggestion
### Control Plane nodes
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+### Worker Nodes
+
+   * `/var/log/kubelet.log` - Kubelet, responsible for running containers on the node
+   * `/var/log/kube-proxy.log` - Kube Proxy, responsible for service load balancing
+
+## A general overview of cluster failure modes
+
+This is an incomplete list of things that could go wrong, and how to adjust your cluster setup to mitigate the problems.
+
+### Root causes:
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:238]:
```suggestion
### Root causes
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+
+  - VM(s) shutdown
+  - Network partition within cluster, or between cluster and users
+  - Crashes in Kubernetes software
+  - Data loss or unavailability of persistent storage (e.g. GCE PD or AWS EBS volume)
+  - Operator error, for example misconfigured Kubernetes software or application software
+
+### Specific scenarios:
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:246]:
```suggestion
### Specific scenarios
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 11]
```diff
@@ -0,0 +1,308 @@
...
+      - node marked unhealthy
+      - replication controllers start new pods elsewhere
+  - Cluster operator error
+    - Results
+      - loss of pods, services, etc
+      - lost of apiserver backing store
+      - users unable to read API
+      - etc.
+
+### Mitigations:
```
Comment (tengqm) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:280]:
```suggestion
### Mitigations
```

**Comment by tengqm** [comment]
/approve
Thanks.

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 12]
```diff
@@ -0,0 +1,308 @@
...
+
+  - VM(s) shutdown
+  - Network partition within cluster, or between cluster and users
+  - Crashes in Kubernetes software
+  - Data loss or unavailability of persistent storage (e.g. GCE PD or AWS EBS volume)
+  - Operator error, for example misconfigured Kubernetes software or application software
+
+### Specific scenarios
+
+  - Apiserver VM shutdown or apiserver crashing
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:248]: nit: write “API server” not “Apiserver”.

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+    - Results
+      - unable to stop, update, or start new pods, services, replication controller
+      - existing pods and services should continue to work normally, unless they depend on the Kubernetes API
+  - Apiserver backing storage lost
+    - Results
+      - apiserver should fail to come up
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:254]:
```suggestion
      - the kube-apiserver component fails to start successfully and become healthy
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+### Worker Nodes
+
+   * `/var/log/kubelet.log` - Kubelet, responsible for running containers on the node
+   * `/var/log/kube-proxy.log` - Kube Proxy, responsible for service load balancing
+
+## A general overview of cluster failure modes
+
+This is an incomplete list of things that could go wrong, and how to adjust your cluster setup to mitigate the problems.
+
+### Root causes
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:238]:
```suggestion
### Contributing causes
```
(I don't want to promote the ingle “root cause” concept)

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+   * `/var/log/kube-apiserver.log` - API Server, responsible for serving the API
+   * `/var/log/kube-scheduler.log` - Scheduler, responsible for making scheduling decisions
+   * `/var/log/kube-controller-manager.log` - Controller that manages replication controllers
+
+### Worker Nodes
+
+   * `/var/log/kubelet.log` - Kubelet, responsible for running containers on the node
+   * `/var/log/kube-proxy.log` - Kube Proxy, responsible for service load balancing
+
+## A general overview of cluster failure modes
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:234]:
```suggestion
## Cluster failure modes
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 16]
```diff
@@ -0,0 +1,308 @@
...
+### Control Plane nodes
+
+   * `/var/log/kube-apiserver.log` - API Server, responsible for serving the API
+   * `/var/log/kube-scheduler.log` - Scheduler, responsible for making scheduling decisions
+   * `/var/log/kube-controller-manager.log` - Controller that manages replication controllers
+
+### Worker Nodes
+
+   * `/var/log/kubelet.log` - Kubelet, responsible for running containers on the node
+   * `/var/log/kube-proxy.log` - Kube Proxy, responsible for service load balancing
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:232]:
```suggestion
   * `/var/log/kubelet.log` - logs from the kubelet, responsible for running containers on the node
   * `/var/log/kube-proxy.log` - logs from `kube-proxy`, which is responsible for directing traffic to Service endpoints
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 17]
```diff
@@ -0,0 +1,308 @@
...
+    operatingSystem: linux
+    osImage: Ubuntu 21.10
+    systemUUID: aa829ca9-73d7-064d-9019-df07404ad448
+```
+
+
+## Looking at logs
+
+For now, digging deeper into the cluster requires logging into the relevant machines.  Here are the locations
+of the relevant log files.  (note that on systemd-based systems, you may need to use `journalctl` instead)
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:221]:
```suggestion
of the relevant log files.  On systemd-based systems, you may need to use `journalctl` instead of
examining log files.
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+
+### Control Plane nodes
+
+   * `/var/log/kube-apiserver.log` - API Server, responsible for serving the API
+   * `/var/log/kube-scheduler.log` - Scheduler, responsible for making scheduling decisions
+   * `/var/log/kube-controller-manager.log` - Controller that manages replication controllers
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:227]:
```suggestion
   * `/var/log/kube-controller-manager.log` - a component that runs most Kubernetes built-in {{<glossary_tooltip text="controllers" term_id="controller">}},
     with the notable exception of scheduling (the kube-scheduler handles scheduling).
```

However, you could just link to the relevant section of https://kubernetes.io/docs/concepts/overview/components/ and use a glossary tooltip for kube-apiserver, kube-scheduler, etc.
eg:
```markdown
* `/var/log/kube-apiserver.log` - {{< glossary_tooltip term_id="kube-apiserver" text="API server" >}} logs
```

File: content/en/docs/tasks/debug/debug-cluster/debug-cluster.md [hunk 19]
```diff
@@ -0,0 +1,308 @@
+---
+reviewers:
+- davidopp
+title: Troubleshooting Clusters
+description: Debugging common cluster issues.
+content_type: concept
+weight: 10
+---
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/debug-cluster.md:8]: This would be good content for `content/en/docs/tasks/debug/debug-cluster/_index.md`

You could then link to the specific pages in the section within the page, and set `no_list: true` in the front matter to suppress the automatic section listing. I recommend you try it and see what you think.

File: content/en/docs/tasks/debug/debug-cluster/_index.md [hunk 20]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Debugging Clusters"
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-cluster/_index.md:2]: How does this section title align with the title of `content/en/docs/tasks/debug/debug-cluster/debug-cluster.md`: “Troubleshooting Clusters”.

I'm thinking that Troubleshooting should be the generic title (not Debugging).

File: content/en/docs/tasks/debug/debug-application/debug-service.md [hunk 21]
```diff
@@ -727,13 +728,13 @@ Service is not working.  Please let us know what is going on, so we can help
...
 Contact us on
-[Slack](/docs/tasks/debug-application-cluster/troubleshooting/#slack) or
+[Slack](/docs/tasks/debug/overview/#slack) or
 [Forum](https://discuss.kubernetes.io) or
 [GitHub](https://github.com/kubernetes/kubernetes).
 
 ## {{% heading "whatsnext" %}}
 
-Visit [troubleshooting document](/docs/tasks/debug-application-cluster/troubleshooting/)
+Visit [troubleshooting document](/docs/tasks/debug/overview/)
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-application/debug-service.md:737]: Can you reword this to make it be idiomatic English?

File: content/en/docs/tasks/debug/overview.md [hunk 22]
```diff
@@ -3,17 +3,19 @@ reviewers:
 - brendandburns
 - davidopp
 content_type: concept
-title: Troubleshooting
+title: Debugging overview
+description: General overview and how to get help.
+weight: 10
 ---
 
 <!-- overview -->
```
Comment (sftim) [inline content/en/docs/tasks/debug/overview.md:11]: I'm not keen on the new page URL and title. I would merge this content into Monitoring, Logging, and Debugging (`content/en/docs/tasks/debug/_index.md`).

File: content/en/docs/tasks/debug/debug-application/_index.md [hunk 23]
```diff
@@ -0,0 +1,5 @@
+---
+title: "Debugging Applications"
+description: Debugging common containerized application issues.
+weight: 20
+---
```
Comment (sftim) [inline content/en/docs/tasks/debug/debug-application/_index.md:5]: If you can, write an overview for the tasks in this section. A couple of sentences would do.

**Review by sftim — COMMENTED** [review]
Thanks for this PR. With some tweaks, this will be a great change.

As it stands I think it's a useful change but I'd prefer to see a few details revised before we merge it, so that we make things easier for localization teams. (If we make the fixes in two goes, that's more work downstream).
